include/turtle/detail/group.hpp

```diff
@@ -22,6 +22,13 @@ namespace detail
     class group
     {
     public:
+        ~group()
+        {
+            for( verifiables_cit it = verifiables_.begin();
+                    it != verifiables_.end(); ++it )
+                (*it)->context_destroyed();
```

Comment:
Is this part of the new singleton implementation or another fix?

Reply:
Both. The new singleton shows or introduces a different initialization and hence finalization order. E. G. A global mocked function gets constructed before the `global` singleton so destruction uses the destroyed singleton